### ISSUE TITLE
Bluetooth: btgatt-server and btgatt-client not connecting to another device simultaneously

### ISSUE THREAD

**Issue by Bhanuswaroophr** [OpenSourceHelpCommunity/Questions#535]
Hi,

I am using btgatt-server and btgatt-client tools of bluez. I can use them and connect to another device like RCU, Mobile. But when I try to run both the tools simultaneously, only one connection can be established. For eg, if i use btgatt-client and connect it to RCU, I can run btgatt-server but it doesnt connect to RCU - it stays at "Started listening on ATT channel. Waiting for connections".
t
Now, if i disconnect the first connection where i used btgatt-client, I can see that server immediately connects to RCU. 
Also, if i run two instances of btgatt-client, each instance connects to Mobile and RCU respectively.

In summary there seems to be no restrictions for two connections as two instances of btgatt clients are possible. Only that two connects where one is trying to connect as a server and another as client is not happeing.

Is there any solution to this? I can see that in server it is waiting in accept call. I believe there is no restriction to do 2 connects one with btgatt client and one with btgatt server.

Unfortunately, I cannot use any other bluez due to the flash and RAM size limitation on my device. Please help if we can fix this issue in btgatt server? 

thanks,
Bhanu
